temporal-sdk/src/main/java/io/temporal/worker/tuning/JVMSystemResourceInfo.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright (C) 2022 Temporal Technologies, Inc. All Rights Reserved.
+ *
+ * Copyright (C) 2012-2016 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Modifications copyright (C) 2017 Uber Technologies, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this material except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.temporal.worker.tuning;
+
+import com.sun.management.OperatingSystemMXBean;
+import io.temporal.common.Experimental;
+import java.lang.management.ManagementFactory;
+import java.lang.management.MemoryMXBean;
+
+/** {@link SystemResourceInfo} implementation that uses JVM-specific APIs to get resource usage. */
+@Experimental
+public class JVMSystemResourceInfo implements SystemResourceInfo {
+  // As of relatively recent Java versions (including backports), this class will properly deal with
+  // containerized environments as well as running on bare metal.
+  // See https://bugs.openjdk.org/browse/JDK-8226575 for more details on which versions the fixes
+  // have been backported to.
+  OperatingSystemMXBean osBean = ManagementFactory.getPlatformMXBean(OperatingSystemMXBean.class);
+  MemoryMXBean memoryBean = ManagementFactory.getMemoryMXBean();
+
+  @Override
+  @SuppressWarnings("deprecation") // deprecated APIs needed since replacements are for Java 14+
+  public double getCpuUsagePercent() {
+    return osBean.getSystemCpuLoad();
+  }
+
+  @Override
+  @SuppressWarnings("deprecation") // deprecated APIs needed since replacements are for Java 14+
+  public double getMemoryUsagePercent() {
+    long jvmUsedMemory =
```

Comment:
Is there a reason you aren't doing `Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()`?

Reply:
Nah, not really. That probably makes sense to do instead.